src/Java.Interop.Tools.JavaCallableWrappers/Resources/MonoRuntimeProvider.Bundled.java

```diff
@@ -19,7 +21,20 @@ public boolean onCreate ()
 	public void attachInfo (android.content.Context context, android.content.pm.ProviderInfo info)
 	{
 		// Mono Runtime Initialization {{{
-		mono.MonoPackageManager.LoadApplication (context, context.getApplicationInfo (), new String[]{context.getApplicationInfo ().sourceDir});
+		ApplicationInfo applicationInfo = context.getApplicationInfo ();
+		String[] apks = null;
+		if (Build.VERSION.SDK_INT >= 21) {
+			String[] splitApks = applicationInfo.splitPublicSourceDirs;
+			if (splitApks != null && splitApks.length > 0) {
+				apks = new String[splitApks.length + 1];
+				apks [0] = applicationInfo.sourceDir;
+				System.arraycopy (splitApks, 0, apks, 1, splitApks.length);
+			}
+		}
+		if (apks == null) {
+			apks = new String[] { applicationInfo.sourceDir };
```

Comment:
Do we know if the app ordering matters here? I would guess not... maybe?

Reply:
That `base.apk` should definitely be first, but...

Let's say you have `libfoo.so` and you include it for `armeabi-v7a` and `arm64-v8a`.

What will it do on a 64-bit device?

This post suggests that there would be two APKs for the two architectures on a device that supports both: https://stackoverflow.com/a/51094667/132442

I think I need to just find a native library like this and deploy an app bundle to a 64-bit device to see what happens.

Reply:
Hmm, I enabled AOT in my test, and my app bundle has both `armeabi-v7a` and `arm64-v8a` so files.

But when I generate the "apk set" for my attached device, it only includes an `arm64-v8a` APK. Same on device.

Here is a zip containing the two files (`.aab` and `.apks`): https://1drv.ms/u/s!Apmn9psh-S8hlGvmXK_vCndyhlxo

Maybe this is not an issue, since Google Play _wants_ to only download/install one abi? The AOT'd app starts up fine.

Reply:
I believe Google Play does optimise and delivers only the abi's needed, so perhaps this is not an issue.

Reply:
Our lowest API level 19 so this won't compile if the user is targeting < 21 ....